app/controllers/internal_api/v1/pillars_controller.rb

```diff
@@ -20,24 +20,21 @@ def pillar_contents
   end
 
   def registry_contents
-    registries = DockerRegistry.is_registry.map do |reg|
-      {
-        url:  reg.url,
-        cert: (reg.certificate.present? ? reg.certificate.certificate : nil)
-      }
-    end
-    registry_mirrors = DockerRegistry.is_mirror.group(:mirror).pluck(:mirror)
-    registry_mirrors.map! do |remote_registry_url|
-      {
-        url:     remote_registry_url,
-        mirrors: DockerRegistry.where(mirror: remote_registry_url).map do |reg|
-          {
-            url:  reg.url,
-            cert: (reg.certificate.present? ? reg.certificate.certificate : nil)
-          }
-        end
-      }
+    registries = []
+    Registry.all.map do |reg|
+      registry ||= {}
+      registry[:url]  = reg.url
+      registry[:cert] = reg.try(:certificate).try(:certificate)
```

Comment:
The first `certificate` should always be present, this can be `reg.certificate.try(:certificate)`.